src/main/java/org/humancellatlas/ingest/state/persistence/AutoLoader.java

```diff
@@ -29,25 +40,98 @@
+            } else {
+                updateStateMachineToEnvelopeState(stateMachine, envelope.get());
+                submissionStateMonitor.monitorSubmissionEnvelope(envelope.get(), stateMachine);
+                String state = stateMachine.getState().getId().toString();
+                log.info(String.format("Restored %s with current state: %s", envelopeUuid.toString(), state));
             }
-        });
+
+        } catch (RuntimeException e) {
+            String desc = String.format("An unexpected error has occurred in loading the state machine for submission %s: %s",
+                    stateMachine.getId(), e.getMessage());
+
+            String action = "Please check the stack trace.";
+
+            handleException(desc, action, e);
+        }
+    }
+
+    private Optional<SubmissionEnvelopeReference> getEnvelope(UUID envelopeUuid) {
+        try {
+            SubmissionEnvelopeReference envelope = ingestApiClient.referenceForSubmissionEnvelope(envelopeUuid);
+            return Optional.of(envelope);
+        } catch (UnrecognisedSubmissionStateException e) {
+            String desc = String.format("A submission state value is not recognised in the state tracker: %s", e.getMessage());
+            String action = "Consider defining a new SubmissionState enum value";
+            handleException(desc, action, e);
+        } catch (HttpClientErrorException.NotFound e) {
+            log.info(String.format("Submission %s was not found.", envelopeUuid.toString()));
+            return Optional.empty();
+        } catch (RestClientException e) {
+            String desc = String.format("An error has occurred in retrieving submission envelope %s : %s",
+                    envelopeUuid.toString(), e.getMessage());
+            String action = "Please ensure Ingest API is running.";
+            handleException(desc, action, e);
+        }
+        return Optional.empty();
+    }
+
+    private StateMachine<SubmissionState, SubmissionEvent> updateStateMachineToEnvelopeState(StateMachine<SubmissionState, SubmissionEvent> stateMachine, SubmissionEnvelopeReference envelope) throws UnrecognisedSubmissionStateException {
+        SubmissionState currentState = stateMachine.getState().getId();
+        SubmissionState correctState = envelope.getState();
+
+        // Note that some states can have extended state variables which tracks some metadata in a submission.
+        // Please check StateMachineConfiguration class to check what these extended state variables are.
+        // Without that extended state, the current state may not be able to successfully be transitioned to the next state
+        // It would be more complicated to sync these extended states from core.
+        // It might be best to redesign how we use the state machines and not have a separate state tracker component outside core.
+
+        if (!currentState.equals(correctState) && !STATES_WITH_EXTENDED_STATE.contains(correctState)) {
```

Comment:
I would hide the implementation here and add a predicate method to `SubmissionState`